crates/pop-contracts/src/up.rs

```diff
@@ -117,3 +120,79 @@ pub async fn instantiate_smart_contract(
 	let instantiate_result = instantiate_exec.instantiate(Some(gas_limit)).await?;
 	Ok(instantiate_result.contract_address.to_string())
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Are we testing for errors as well?

Reply:
Not everything in this file is tested; unit testing for uploading a smart contract is more complex. However, custom errors are tested here: https://github.com/r0gue-io/pop-cli/blob/alex/dry-run-flag/crates/pop-contracts/src/up.rs#L192